src/core/evaluator.js

```diff
@@ -381,10 +381,21 @@ class PartialEvaluator {
 
   async fetchStandardFontData(name) {
     const standardFontNameToFileName = getStdFontNameToFileMap();
+    const filename = standardFontNameToFileName[name];
+    if (this.options.standardFontDataWorkerFetch) {
+      const url = `${this.options.standardFontDataUrl}${filename}.pfb`;
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      const data = new Uint8Array(await response.arrayBuffer());
+      return new Stream(data);
```

Comment:
I believe that you'll need to wrap all of this is a `try...catch` block, and fallback to main-thread data loading if there's any errors, since otherwise there's a very real risk that font data loading will break in GENERIC-builds.

Given that, in the API, you're setting the option to *by default* (in browsers) try and load font data in the worker-thread, there could thus be errors if people have implemented a *custom* `StandardFontDataFactory` e.g. to deal with some CORS issue, or if the PDF.js library is used in an environment and/or with a protocol where `fetch` isn't properly supported.

Reply:
I've been going back and forth on this, but I'm leaning towards not automatically falling back to getting data on the main thread if fetch fails as it really is an error that the user of the library should handle either by disabling fetch or putting the font files in the correct location. Also, if we fallback automatically this may silently regress and fetch in the worker has been supported for quite awhile now.